### PR TITLE
Ensure right joycon motion data is set

### PR DESCRIPTION
**Pull request by emmauss**
There's a bug where the right joycon is not retrieved from cemuhook, and that the right joycon motion shared mem is cleared after update. This aims to fix that issue. 
This should allow right joycon to work in Skyward Sword.

### CODE REVIEW

File: Ryujinx.Input/HLE/NpadController.cs [hunk 1]
```diff
@@ -310,24 +311,31 @@ public void Update()
                         // First of all ensure we are registered
                         _cemuHookClient.RegisterClient(clientId, cemuControllerConfig.DsuServerHost, cemuControllerConfig.DsuServerPort);
 
-                        // Then request data
+                        // Then request and retrive the data
```
Comment (Shahil-Ayato) [inline Ryujinx.Input/HLE/NpadController.cs:314]: `retrive` should be `retrieve`

**Comment by gdkchan** [comment]
Closes #2127 I guess?

File: Ryujinx.Input/HLE/NpadManager.cs [hunk 2]
```diff
@@ -179,7 +182,9 @@ public void Update(float aspectRatio = 0)
 
                         inputState.Buttons |= _device.Hid.UpdateStickButtons(inputState.LStick, inputState.RStick);
 
-                        motionState = controller.GetHLEMotionState();
+                        isJoyconPair = inputConfig.ControllerType == Common.Configuration.Hid.ControllerType.JoyconPair;
+
+                        motionState = (controller.GetHLEMotionState(), isJoyconPair ? controller.GetHLEMotionState(true) : default);
```
Comment (Thog) [inline Ryujinx.Input/HLE/NpadManager.cs:187]: Note really a huge fan of ternary inside tuples

**Review by Thog — APPROVED** [review]
lgtm only one thing to change

File: Ryujinx.Input/HLE/NpadManager.cs (hunk 2 above)
Comment (emmauss) [reply Ryujinx.Input/HLE/NpadManager.cs]: fixed

File: Ryujinx.Input/HLE/NpadController.cs [hunk 3]
```diff
@@ -395,20 +407,32 @@ public GamepadInput GetHLEInputState()
...
-        public SixAxisInput GetHLEMotionState()
+        public SixAxisInput GetHLEMotionState(bool isSecondaryController = false)
         {
             float[] orientationForHLE = new float[9];
             Vector3 gyroscope;
             Vector3 accelerometer;
             Vector3 rotation;
 
-            if (_motionInput != null)
+            var motionInput = _mainMotionInput;
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:417]:
```suggestion
            MotionInput motionInput = _mainMotionInput;
```

File: Ryujinx.Input/HLE/NpadController.cs [hunk 4]
```diff
@@ -395,20 +407,32 @@ public GamepadInput GetHLEInputState()
...
+
+            if(isSecondaryController)
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:419]:
```suggestion
            if (isSecondaryController)
```

File: Ryujinx.Input/HLE/NpadController.cs [hunk 5]
```diff
@@ -395,20 +407,32 @@ public GamepadInput GetHLEInputState()
             return state;
         }
 
-        public SixAxisInput GetHLEMotionState()
+        public SixAxisInput GetHLEMotionState(bool isSecondaryController = false)
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:410]:
```suggestion
        public SixAxisInput GetHLEMotionState(bool isJoyconPair = false)
```

File: Ryujinx.Input/HLE/NpadController.cs [hunk 6]
```diff
@@ -395,20 +407,32 @@ public GamepadInput GetHLEInputState()
...
             Vector3 gyroscope;
             Vector3 accelerometer;
             Vector3 rotation;
 
-            if (_motionInput != null)
+            var motionInput = _mainMotionInput;
+
+            if(isSecondaryController)
+            {
+                if(_secondaryMotionInput == null)
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadController.cs:421]:
```suggestion
                if (_secondaryMotionInput == null)
```

File: Ryujinx.Input/HLE/NpadManager.cs [hunk 7]
```diff
@@ -189,14 +196,21 @@ public void Update(float aspectRatio = 0)
...
 
                     inputState.PlayerId = (Ryujinx.HLE.HOS.Services.Hid.PlayerIndex)inputConfig.PlayerIndex;
-                    motionState.PlayerId = (Ryujinx.HLE.HOS.Services.Hid.PlayerIndex)inputConfig.PlayerIndex;
+                    motionState.Item1.PlayerId = (Ryujinx.HLE.HOS.Services.Hid.PlayerIndex)inputConfig.PlayerIndex;
 
                     hleInputStates.Add(inputState);
-                    hleMotionStates.Add(motionState);
+                    hleMotionStates.Add(motionState.Item1);
+
+                    if(isJoyconPair && !motionState.Item2.Equals(default))
```
Comment (AcK77) [inline Ryujinx.Input/HLE/NpadManager.cs:208]:
```suggestion
                    if (isJoyconPair && !motionState.Item2.Equals(default))
```

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Comment by emmauss** [comment]
Addressed comments

**Comment by emmauss** [comment]
Addressed comments

**Review by AcK77 — APPROVED** [review]

**Review by gdkchan — APPROVED** [review]
Looks alright, please add `Fixes #2127` if the issue is fixed by this change on the PR description, to have it auto-close on merge.

**Comment by emmauss** [comment]
> 
> 
> Looks alright, please add `Fixes #2127` if the issue is fixed by this change on the PR description, to have it auto-close on merge.

It fixes only the last comments on #2127 . It doesn't fix the issue in the first comment.

**Pull request merged**